manager/assets/modext/core/modx.layout.js

```diff
@@ -52,13 +50,22 @@ MODx.Layout = function(config){
 };
 Ext.extend(MODx.Layout, Ext.Viewport, {
     /**
+     * Initialize the state provider to use
+     */
+    initStateProvider: function() {
+        var provider = MODx.config.stateprovider || MODx.HttpProvider,
+            sp = new provider();
```

Comment:
We might want to make sure `provider` is an available class, else fallback to default `MODx.HttpProvider`